src/hardware/mame/emu.h

```diff
@@ -18,9 +18,7 @@
 #include <stdarg.h>
 #endif
 
-#ifndef M_PI
-#define M_PI           3.14159265358979323846
```

Comment:
When I read this code for the first time my immediate reaction was: "why original author redefined M_PI?" (perhaps for the same reason why someone defined PI in `dbopl.cpp` file).

And it seems like the reason was usual… MSVC.

Turns out, Microsoft's compiler does not export macros with constants by default when including `cmath` nor `math.h`… so original authors of this code probably thought M_PI simply does not exist on Windows.

However: https://docs.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=vs-2019

I think we should place `#define _USE_MATH_DEFINES` in `src/platform/visualc/config.h` (with appropriate comment explaining why); `config.h` file is (should) already be included before `cmath` (as long as our include order rules are followed).

Having this change in a separate PR merged earlier should make us double-sure the change works ok across our various build environments :)

Also, I'm not sure if it's relevant for furure PRs for PC speaker, but there are other immeasurable constants that can be nice to use, like: `M_2_PI` or `M_2_SQRTPI`.

Reply:
Sounds good enough; so we'll enable MSVC's math constants by default, and then use `M_PI` throughout?

For cases where we want to PI with `float` values, should we locally assign `constexpr float f_pi = static_cast<float>(M_PI);`?

Reply:
> Sounds good enough; so we'll enable MSVC's math constants by default, and then use `M_PI` throughout?

Yup :)

> For the audio code using 32-bit calculations, should we locally assign `constexpr float f_pi = static_cast<float>(M_PI)` as needed?

I don't think so. Let's just use M_PI everywhere - I doubt it will cause any problems. Note: `fltype` used for casting in mame code is actually `double`.

> There are currently only three areas we need it as a float: the DC Silencer, fast sine & cosine approximation functions, and GUS pan-angle constant-power calculation.

Do we *really* need 32-bit floating-point type in any of those areas? I think `double` will be just fine in all of them.

Reply:
It's the right approach; they run an extra pass over all the audio samples (on-top of the existing 2 x 44,100 calcs/second); the math only has to retain rounding errors withing 16-bit; so going all the way up to 64-bit doubles and then casting back down to 16bit is pretty wasteful; plus, given the goal is 32-bit floating points, SIMD will likely be used (it should be able to do left and right channel in the same instruction, because they do not depend on eachother
).

Reply:
@kcgen if you feel in some contexts using `constexpr float` for storing pi is needed, then go ahead :)

(it seems like build failed, because some Windows code still needs to be updated to use M_PI in place of PI macro)

Reply:
I missed it in fpu.h; repushed and all passing again. Nice easy solution @dreamer, thanks for the suggestion!